Sources/News.php

```diff
@@ -1849,4 +2129,758 @@ function getXmlProfile($xml_format)
 	return $data;
 }
 
+/**
+ * Get a user's posts.
+ * The returned array will be generated to match the xml_format.
+ *
+ * @param string $xml_format The XML format. Can be 'atom', 'rdf', 'rss', 'rss2' or 'smf'
+ * @return array An array of arrays containing data for the feed. Each array has keys corresponding to the appropriate tags for the specified format.
+ */
+function getXmlPosts($xml_format)
+{
+	global $scripturl, $modSettings, $board, $txt, $context, $user_info;
+	global $query_this_board, $smcFunc, $sourcedir, $cachedir;
+
+	$uid = isset($_GET['u']) ? (int) $_GET['u'] : $user_info['id'];
+
+	if (empty($uid) || (!allowedTo('profile_view') && $uid != $user_info['id']))
+		return array();
+
+	$show_all = $user_info['is_admin'] || $uid == $user_info['id'];
+
+	// You are allowed in this special case to see your own posts from anywhere
+	if ($show_all)
+		$query_this_board = preg_replace('/\{query_see_board\}\s*(AND )?/', '', $query_this_board);
```

Comment:
Just a little comment from my side on this line and only because I've passed through that suggestion and got it turned down. Last time, giving people access to their posts after they lost access to them was a big **no-go** and for kind of good security and privacy reasons (and I'm pretty sure sm.org is a good example of the "bad habit" of editing posts pinned ages ago from team members long gone adding any kind of info to them, maybe including passwords, or personal data of someone else for taking notes, etc.).
Maybe things have changed, but I kind of though it was worth mentioning, in case for example @Kindred-999 wasn't aware of the change. :stuck_out_tongue: